test/logging_unittest.cc

```diff
@@ -0,0 +1,26 @@
+#include "../src/logging.h"
+#include "gtest/gtest.h"
+
+#include <regex>
+#include <sstream>
+
+namespace google {
+namespace {
+
+TEST(LoggingTest, Logger) {
+  std::ostringstream out;
+
+  // Logger flushes on destruction.
+  {
+    LogStream stream(out);
+    Logger logger("somefile.cc", 123, Logger::WARNING, &stream);
+    logger << "Test message";
+  }
+
+  EXPECT_TRUE(std::regex_match(out.str(), std::regex(
+     "W\\d{4} \\d{2}:\\d{2}:\\d{2} 0x[0-9a-f]{12} "
```

Comment:
Should we be so strict on the date format, instead of just validating the W, the file, the line, and the message?

Reply:
I'm leaning towards having a strict check so the log format we produce is predictable and can potentially be parsed into structured logs easier if needed.